src/python/pants/build_graph/address.py

```diff
@@ -388,30 +389,24 @@ def sanitize(s: str) -> str:
         prefix = sanitize(self.spec_path)
         return f"{prefix}{path}{target}{params}{generated}"
 
+    def parametrize(self, parameters: Mapping[str, str]) -> Address:
+        """Creates a new Address with the given `parameters` merged over self.parameters."""
+        merged_parameters = {**self.parameters, **parameters}
+        return self.__class__(
+            self.spec_path,
+            target_name=self._target_name,
+            generated_name=self.generated_name,
+            relative_file_path=self._relative_file_path,
+            parameters=merged_parameters,
+        )
+
     def maybe_convert_to_target_generator(self) -> Address:
```

Comment:
Is `target_generator` a misnomer? Maybe `base_target`?

Reply:
Yea, it probably is. It used to be `build_target`, although I like `base`.

Reply:
Probably best as a followup though?

Reply:
Fine with me.

Maybe open an issue tracking all the little fixes like this? I know you have some dedicated tickets for bigger changes.

Reply:
Will do.